lib/node_modules/@stdlib/napi/argv-complex64/src/main.c

```diff
@@ -0,0 +1,117 @@
+*/
+
+#include "stdlib/napi/argv_complex64.h"
+#include "stdlib/assert/napi/status_ok.h"
+#include "stdlib/assert/napi/is_type.h"
+#include "stdlib/complex/float32.h"
+#include <node_api.h>
+
+/**
+* Converts a Node-API value to a single-precision complex floating-point number.
+*
+* @param env       environment under which the function is invoked
+* @param value     Node-API value
+* @param out       destination for storing output value
+* @param message   error message
+* @param err       pointer for storing a JavaScript error
+* @return          status code indicating success or failure (returns `napi_ok` if success)
+*
+* @example
+* #include "stdlib/napi/argv_complex64.h"
+* #include "stdlib/complex/float32.h"
+* #include <node_api.h>
+*
+* static napi_value addon( napi_env env, napi_callback_info info ) {
+*     napi_value value;
+*
+*     // ...
+*
+*     stdlib_complex64_t out;
+*     napi_value err;
+*     napi_status status = stdlib_napi_argv_complex64( env, value, &out, "Must be a Complex64 number.", &err );
+*     assert( status == napi_ok );
+*     if ( err != NULL ) {
+*         assert( napi_throw( env, err ) == napi_ok );
+*         return NULL;
+*     }
+*
+*     // ...
+* }
+*/
+napi_status stdlib_napi_argv_complex64( const napi_env env, const napi_value value, stdlib_complex64_t *out, const char *message, napi_value *err ) {
+
+	stdlib_assert_napi_value_is_type( env, value, napi_object, message, err );
+	if ( *err != NULL ) {
+		return napi_ok;
+	}
+
+	bool hprop;
+	status = napi_has_named_property( env, value, "re", &hprop );
+	assert( status == napi_ok );
+	if ( !hprop ) {
+		status = napi_throw_type_error( env, NULL, "Invalid argument. The Complex64 object must have a real component." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+	
+	napi_value re0;
+	status = napi_get_named_property( env, value, "re", &re0 );
+	assert( status == napi_ok );
+
+	napi_valuetype retype;
+	status = napi_typeof( env, re0, &retype );
+	assert( status == napi_ok );
+	if ( retype != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "Invalid argument. The Complex64 object must have a real component which is a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	status = napi_has_named_property( env, value, "im", &hprop );
+	assert( status == napi_ok );
+	if ( !hprop ) {
+		status = napi_throw_type_error( env, NULL, "Invalid argument. The Complex64 object must have an imaginary component." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_value im0;
+	status = napi_get_named_property( env, argv[ 0 ], "im", &im0 );
+	assert( status == napi_ok );
+
+	napi_valuetype imtype;
+	status = napi_typeof( env, im0, &imtype );
+	assert( status == napi_ok );
+	if ( imtype != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "Invalid argument. The Complex64 object must have an imaginary component which a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	single real;
```

Comment:
It is clear to me that you did not actually test this implementation. The C type should be `float`. Similarly, there is no `napi_get_value_single` API.

Reply:
@kgryte This was my first PR and on that time I'm not aware lots of thing. After that I'm not go through it now I will revisit it and change accordingly.